app/src/pages/playground/fetchPlaygroundPrompt.ts

```diff
@@ -95,6 +151,65 @@ export const promptVersionToInstance = (
   } satisfies Partial<PlaygroundInstance>;
 };
 
+/**
+ * Converts invocation parameters to an object of key-value pairs, where the key is the invocation parameter name
+ * and the value is the value of the invocation parameter.
+ *
+ * @param invocationParameters - The invocation parameters set in the instance
+ * @param supportedInvocationParameters - The supported invocation parameters for the model of the instance
+ *
+ * @returns The invocation parameters as an object, constrained to the supported invocation parameters
+ */
+export const invocationParametersToObject = (
+  invocationParameters: PlaygroundInstance["model"]["invocationParameters"],
+  supportedInvocationParameters: PlaygroundInstance["model"]["supportedInvocationParameters"]
+) => {
+  const invocationParameterDefinitionMap =
+    supportedInvocationParameters.length > 0
+      ? supportedInvocationParameters.reduce(
+          (acc, curr) => {
+            if (curr.canonicalName || curr.invocationName) {
+              acc[(curr.canonicalName || curr.invocationName) as string] = curr;
+            }
+            return acc;
+          },
+          {} as Record<string, (typeof supportedInvocationParameters)[number]>
+        )
+      : {};
+  return invocationParameters.reduce(
+    (acc, curr) => {
+      const definition =
+        invocationParameterDefinitionMap[
+          curr.canonicalName || curr.invocationName
+        ];
+      if (definition) {
+        acc[curr.invocationName] =
+          curr[
+            toCamelCase(
+              definition.invocationInputField as string
+            ) as keyof typeof curr
+          ];
+      }
+      return acc;
+    },
+    {} as Record<
+      string,
+      string | number | boolean | null | Record<string, unknown> | unknown[]
+    >
+  );
+};
+
+const HIDDEN_INVOCATION_PARAMETERS = [
+  {
```

Comment:
don't we want the tool choice?

Reply:
Yes, fixed. This revealed another bug where I was not consolidating some invocation param parsing. This meant that the prompt picker on the playground page would not carry over invocation params or tool choice.

Now everything has been consolidated and the issue is fixed. If you use `fetchPlaygroundPromptAsInstance()` you are now getting a full playground instance without any post-processing needed.